.github/workflows/tests.yml

```diff
@@ -11,134 +11,73 @@ jobs:
-        run: poetry install --no-interaction --no-root
-
       - name: Run pytest
         run: poetry run pytest
 
   mypy:
     runs-on: ubuntu-latest
 
     steps:
-      - name: Check out repository
-        uses: actions/checkout@v3
-
-      - name: Set up Python
-        id: setup-python
-        uses: actions/setup-python@v4
+      - name: Install Python Project
+        uses: CSSUoB/action-install-python-project@v1.0.2
         with:
           python-version: 3.11
 
-      - name: Install Poetry
-        uses: snok/install-poetry@v1
-        with:
-          virtualenvs-create: true
-          virtualenvs-in-project: true
-          installer-parallel: true
-
-      - name: Load cached venv
-        id: cached-poetry-dependencies
-        uses: actions/cache@v3
-        with:
-          path: .venv
-          key: venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/poetry.lock') }}
-
-      - name: Install dependencies
-        if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
-        run: poetry install --no-interaction --no-root
-
       - name: Run mypy
-        run: poetry run mypy .
+        run: poetry run mypy .; poetry run mypy ./.github/workflows/scripts
 
   ruff:
     runs-on: ubuntu-latest
 
     steps:
-      - name: Check out repository
-        uses: actions/checkout@v3
-
-      - name: Set up Python
-        id: setup-python
-        uses: actions/setup-python@v4
+      - name: Install Python Project
+        uses: CSSUoB/action-install-python-project@v1.0.2
         with:
           python-version: 3.11
 
-      - name: Install Poetry
-        uses: snok/install-poetry@v1
-        with:
-          virtualenvs-create: true
-          virtualenvs-in-project: true
-          installer-parallel: true
-
-      - name: Load cached venv
-        id: cached-poetry-dependencies
-        uses: actions/cache@v3
-        with:
-          path: .venv
-          key: venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/poetry.lock') }}
-
-      - name: Install dependencies
-        if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
-        run: poetry install --no-interaction --no-root
-
       - name: Run ruff
         run: poetry run ruff check . --no-fix --extend-select TD002,TD003
 
   poetry-check:
     runs-on: ubuntu-latest
 
+    steps:
+      - name: Install Python Project
+        uses: CSSUoB/action-install-python-project@v1.0.2
+        with:
+          python-version: 3.11
+
+      - name: Run poetry check
+        run: poetry check --no-interaction --lock && poetry check --no-interaction
+
+  json-check:
+    runs-on: ubuntu-latest
+
     steps:
       - name: Check out repository
         uses: actions/checkout@v3
 
       - name: Set up Python
-        id: setup-python
         uses: actions/setup-python@v4
         with:
-          python-version: 3.11
+          python-version: "3.x"
```

Comment:
If this is being changed to `3.x`, should the other uses of `3.11` also be changed?

Reply:
All the other workflows are pinned at a specific Python version because that is the only specific version supported by our code & the dependencies. In the other workflows the code is actually run/imported so the version matching has to be correct.

However, the JSON checker does not run/import any of our code (hence why the `CSSUoB/action-install-python-project` action is not used). Instead this workflow just statically checks any JSON files using Python's in-built JSON linter which is standardised across all Python 3 versions. Therefore the specific minor Python version does not matter.